src/components/EditTrip.js

```diff
@@ -1,18 +1,222 @@
 import "./styles/EditTrip.css";
-import { useHistory } from "react-router";
+import { useHistory, useParams } from "react-router";
+import { Link } from "react-router-dom";
+import {
+  getSingleTripFromLocalStorage,
+  editSingleTripFromLocalStorage,
+} from "../services/myTripsStorage";
+import { useState, useEffect } from "react";
 
 export default function EditTrip() {
   const history = useHistory();
+  const { id } = useParams();
+
+  const [inputDestinationName, setInputDestinationName] = useState("");
+  const [inputTripStart, setInputTripStart] = useState("");
+  const [inputTripEnd, setInputTripEnd] = useState("");
+  const [inputTransportType, setInputTransportType] = useState("");
+  const [inputTripDeparture, setInputTripDeparture] = useState("");
+  const [inputTripArrival, setInputTripArrival] = useState("");
+  const [inputTripAccommodation, setInputTripAccommodation] = useState("");
+  const [inputCheckinDate, setInputCheckinDate] = useState("");
+  const [inputCheckinTime, setInputCheckinTime] = useState("");
+  const [inputCheckoutDate, setInputCheckoutDate] = useState("");
+  const [inputCheckoutTime, setInputCheckoutTime] = useState("");
+  const [inputSightseeing, setInputSightseeing] = useState("");
+
+  function handleOnSubmit(e) {
+    e.preventDefault();
+
+    editSingleTripFromLocalStorage(id, {
+      name: inputDestinationName,
+      start: inputTripStart,
+      end: inputTripEnd,
+      transportation: inputTransportType,
+      departure: inputTripDeparture,
+      arrival: inputTripArrival,
+      accommodation: inputTripAccommodation,
+      checkinDate: inputCheckinDate,
+      checkinTime: inputCheckinTime,
+      checkoutDate: inputCheckoutDate,
+      checkoutTime: inputCheckoutTime,
+      sightseeing: inputSightseeing,
+    });
+    alert(`Trip to ${inputDestinationName} edited!`);
+    history.push("/myTrips");
+  }
+
+  useEffect(() => {
+    const myTrip = getSingleTripFromLocalStorage(id);
+    setInputDestinationName(myTrip.name);
+    setInputTripStart(myTrip.start);
+    setInputTripEnd(myTrip.end);
+    setInputTransportType(myTrip.transportation);
+    setInputTripDeparture(myTrip.departure);
+    setInputTripArrival(myTrip.arrival);
+    setInputTripAccommodation(myTrip.accommodation);
+    setInputCheckinDate(myTrip.checkinDate);
+    setInputCheckinTime(myTrip.checkinTime);
+    setInputCheckoutDate(myTrip.checkoutDate);
+    setInputCheckoutTime(myTrip.checkinTime);
+    setInputSightseeing(myTrip.sightseeing);
+  }, [id]);
+
   return (
-    <div className="EditTrip">
-      <h2>Edit Page</h2>
-      <button
-        onClick={() => {
-          history.goBack();
-        }}
-      >
-        Return
-      </button>
+    <div className="NewTripForm">
```

Comment:
It seems like this whole form is the same for editing and for creating. 
This is something you could refactor into a new component and use it in both places.

The component could be used like this:
```js
<TripForm trip={objectContainingATrip} onSubmit={functionToHandleSubmit} />
```

The `TripForm` could have its own state to manage the inputs, and receive just the "initial trip data" as a prop. If no initial trip data is defined, then it could assume it's to create a new one.

This might be a larger task, you could add it to your board as a refactor task that you can do later.

Reply:
Thanks Namir. I created a new issue to make a new component for the form as soon as I finish my CSS